test/api/support/ffactory.js

```diff
@@ -0,0 +1,98 @@
+const Promise = require("bluebird")
```

Comment:
I'm not excited about using a proprietary Promise to get some syntactic sugar, but I'm not going to be dogmatic about it. This chunk of code is pretty meta, so making it easier to read is a good thing.